### DISCUSSION TITLE
Ganache: don't confuse network and chain ID

### DISCUSSION THREAD

**Discussion by n4n0b1t3** [smartcontractkit/full-blockchain-solidity-course-py#916] [General]
Whenever the terms Network ID and Chain ID are used without distinction, it should be noted that both IDs can be different for a server such as Ganache. Thus, in the tutorial at this point here 
[Solidity, Blockchain, and Smart Contract Course 3:55:09](https://youtu.be/M576WGiDBdQ?t=14109)
the network ID and NOT the chain id is copied. As you can see here, my Ganache is using different IDs.

```
>>> from web3 import Web3
>>> w3 = Web3(Web3.HTTPProvider("http://127.0.0.1:8545"))
>>> w3.eth.chain_id
1337
>>> w3.net.version
'5777'
>>> 
```

I suggest setting the chain_id programmatically in the source code as follows so that no errors occur.
`"chainId": w3.eth.chain_id, `

Here is a good article that also says that it is better not to use network_ids.
[ChainId vs NetworkId? How do they differ on Ethereum?](https://medium.com/@pedrouid/chainid-vs-networkid-how-do-they-differ-on-ethereum-eec2ed41635b)

**Comment by cromewar** [comment]
And that's why some people are getting different numbers when deploying, you are amazing @n4n0b1t3 !!
Actually, could you make a PR on [chronological-issues-from-video.md](https://github.com/smartcontractkit/full-blockchain-solidity-course-py/blob/main/chronological-issues-from-video.md)? this could be very helpful.

**Reply by n4n0b1t3** [reply]
OK, proposal is ready :-)
